modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,328 @@
+
+    @BeforeClass
+    public void setUp() {
+
+        application = new Application("rapidExperimentApplication");
+
+        experimentsListOfRapidApplication = getApplicationExperiments(application);
+        pauseTerminateAndDeleteExperiments(experimentsListOfRapidApplication);
+
+        rapidExperiment1 = ExperimentFactory.createExperiment("_1" + UUID.randomUUID().toString(), -1)
+                .setApplication(application).setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+        rapidExperiment2 = ExperimentFactory.createExperiment("_2" + UUID.randomUUID().toString(), -1)
+                .setApplication(application);
+
+        rapidExperiment3 = ExperimentFactory.createExperiment("_3" + UUID.randomUUID().toString(), -1)
+                .setApplication(application).setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+
+        rapidExperiment1 = postExperiment(rapidExperiment1);
+        rapidExperiment2 = postExperiment(rapidExperiment2);
+        rapidExperiment3 = postExperiment(rapidExperiment3);
+        expList.add(rapidExperiment1);
+        expList.add(rapidExperiment2);
+        expList.add(rapidExperiment3);
+        for (int i = 1; i <= 3; i++) {
+            Experiment exp = expList.get(i - 1);
+            List<Bucket> bucketList = BucketFactory.createBuckets(exp, 3);
+            postBuckets(bucketList);
+
+            exp.state = Constants.EXPERIMENT_STATE_RUNNING;
+            exp = putExperiment(exp);
+
+        }
+
+    }
+
+    @Test
+    public void testRapidExperiment() {
+
+        Experiment exp = getExperiment(expList.get(0));
+        for (int i = 1; i <= 5; i++) {
+            // lets do an assignment for a user with context set to PROD
+            Assignment assignment = postAssignment(exp, new User("user" + i), "PROD");
+
+            // lets assert the status and response code and also the state of the experiment
+            assertReturnCode(response, HttpStatus.SC_OK);
+            Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+            Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+        }
+
+        // since the experiment crossed the maxusers the experiment should change to PUASED
+        Assignment assignment = postAssignment(exp, new User("user10"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        assignment = postAssignment(exp, new User("user11"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        assignment = postAssignment(exp, new User("user12"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        exp = getExperiment(exp);
+        Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+        expList.set(0, rapidExperiment1);
```

Comment:
Now this code is refactored we no longer need this as each test is made atomic...